obsreport/obsreporttest/otelreporttest.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package obsreporttest // import "go.opentelemetry.io/collector/obsreport/obsreporttest"
+
+import (
+	"fmt"
+	"math"
+	"net/http"
+	"net/http/httptest"
+
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"go.opentelemetry.io/otel/attribute"
+	"go.uber.org/multierr"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+// PrometheusChecker is used to assert exported metrics from a prometheus handler.
+type PrometheusChecker struct {
+	promHandler http.Handler
+}
```

Comment:
I would not want to expose public something tied to Prometheus.

Reply:
Have moved everything behind the existing checks and created a `obsreporttest.SetupOpenTelemetry()` which creates an instance of `TestTelemetry` that will use this `prometheusChecker` to validate the created metrics.